diff --git a/gigaam/__init__.py b/gigaam/__init__.py
index d0b7c89..3819bed 100644
--- a/gigaam/__init__.py
+++ b/gigaam/__init__.py
@@ -126 +126,2 @@ def hash_path(ckpt_path: str) -> str:
-    return hashlib.md5(open(ckpt_path, "rb").read()).hexdigest()
+    with open(ckpt_path, "rb") as checkpoint:
+        return hashlib.file_digest(checkpoint, "md5").hexdigest()
@@ -227 +228 @@ def load_model(
-        checkpoint = torch.load(model_path, map_location="cpu", weights_only=False)
+        checkpoint = torch.load(model_path, map_location="cpu", weights_only=False, mmap=True)
